llvm/lib/Target/AMDGPU/SIFoldOperands.cpp

```diff
@@ -931,7 +931,9 @@ static MachineOperand *lookUpCopyChain(const SIInstrInfo &TII,
   for (MachineInstr *SubDef = MRI.getVRegDef(SrcReg);
        SubDef && TII.isFoldableCopy(*SubDef);
        SubDef = MRI.getVRegDef(Sub->getReg())) {
-    MachineOperand &SrcOp = SubDef->getOperand(1);
+    unsigned SrcIdx = TII.getFoldableCopySrcIdx(*SubDef);
+    MachineOperand &SrcOp = SubDef->getOperand(SrcIdx);
```

Comment:
use getNetNamedOperandIdx? Also probably should just fix the instruction to keep this in index 1

Reply:
The case that will fall is COPY but you can just special case that one

Reply:
I thought a helper is better since people might forget about this special case when they are adding new code. But if you think maintaining the switch is more costly, I'll create follow up patch to remove it.